### PR TITLE
Automatic formatting checks

### PR DESCRIPTION
**Pull request by zmoon**
* Introduce some automatic formatting using [pre-commit](https://pre-commit.com/):
  - [findent](https://www.ratrabbit.nl/ratrabbit/findent/) with `-i4` setting (4 spaces)
  - trim trailing whitespace + check files end with new line character
  - run pre-commit automatically in new GHA workflow

### CODE REVIEW

**Comment by zmoon** [comment]
@drnimbusrain see if you like this style. findent does have some knobs we could turn.

Also I want to add some brief instructions for how to set it up on your local machine for development.

**Comment by drnimbusrain** [comment]
@zmoon thanks, I will review it.  But, can we do this PR to develop instead of main?

File: README.md [hunk 1]
```diff
@@ -18,22 +18,31 @@ Current Canopy-App components:
...
 Massman, W. J., J.M. Forthofer, and M.A. Finney. (2017). An improved canopy wind model for predicting wind adjustment factors and wildland fire behavior. Canadian Journal of Forest Research. 47(5): 594-603. https://doi.org/10.1139/cjfr-2016-0354
 
 Makar, P., Staebler, R., Akingunola, A. et al. The effects of forest canopy shading and turbulence on boundary layer ozone. Nat Commun 8, 15243 (2017). https://doi.org/10.1038/ncomms15243
+
+
+## Development
+
+To set up the pre-commit hooks, invoke
+```
+pre-commit install --install-hooks
```
Comment (drnimbusrain) [inline README.md:46]: @zmoon Does the developer really need to do something before commit to use this, and where exactly is this done?  If, so, can you add a few more details here?

Comment (zmoon) [reply README.md:46]: Yes I can add more details. This step only has to be done once and then it automatically runs the checks/formatters before each commit for you.

**Review by drnimbusrain — APPROVED** [review]

**Pull request merged**